integrationtest/spring/application/src/main/java/org/hibernate/search/integrationtest/spring/repackaged/application/Config.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.spring.repackaged.application;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+
+import org.hibernate.search.mapper.orm.Search;
+
+import acme.org.hibernate.search.integrationtest.spring.repackaged.model.MyEntity;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EntityScan({ "acme.org.hibernate.search.integrationtest.spring.repackaged.model" })
+public class Config {
+
+	@Bean
+	public TestHibernateSearch testHibernateSearch(EntityManagerFactory entityManagerFactory) {
+		return new TestHibernateSearch( entityManagerFactory );
+	}
+
+	public static class TestHibernateSearch {
+		final EntityManagerFactory entityManagerFactory;
+
+		public TestHibernateSearch(EntityManagerFactory entityManagerFactory) {
+			this.entityManagerFactory = entityManagerFactory;
+		}
+
+		@PostConstruct
+		public void check() {
+			try ( EntityManager entityManager = entityManagerFactory.createEntityManager(); ) {
+				EntityTransaction transaction = entityManager.getTransaction();
+				transaction.begin();
+				Search.session( entityManager )
+						.search( MyEntity.class )
+						.select( f -> f.id() )
+						.where( f -> f.matchAll() )
+						.fetchTotalHitCount();
+				transaction.commit();
+				System.out.println( "Hibernate Search read the nested JAR." );
+			}
+		}
+	}
```

Comment:
Is this actually executed in the build?

Reply:
OK, an updated version 😃 
I've added a maven exec that runs this jar -- since the way it's written, it would start the context and shutdown; we can execute it just from the maven build itself... I also added a simple empty spring test to this app module that would just make sure that the context is correctly loaded.